components/hero-banner.php

```diff
@@ -162,139 +162,409 @@ function mm_hero_banner_shortcode( $atts, $content = null, $tag ) {
+                    __( 'Video', 'mm-components ') => 'video',
+                ),
+            ),
+            array(
+                'type' => 'vc_link',
+                'heading' => __( 'Button URL', 'mm-components' ),
+                'param_name' => 'button_link',
+                'dependency' => array(
+                    'element' => 'button_type',
+                    'value' => array(
+                        'standard',
+                    ),
+                ),
+            ),
+            array(
+                'type' => 'textfield',
+                'heading' => __( 'Video URL', 'mm-components' ),
+                'param_name' => 'button_video_url',
+                'dependency' => array(
+                    'element' => 'button_type',
+                    'value' => array(
+                        'video',
+                    ),
+                ),
+            ),
+            array(
+                'type' => 'textfield',
+                'heading' => __( 'Button Text', 'mm-components' ),
+                'param_name' => 'button_text',
+            ),
+            array(
+                'type' => 'dropdown',
+                'heading' => __( 'Button Style', 'mm-components' ),
+                'param_name' => 'button_style',
+                'value' => array(
+                    __( 'Default (solid)', 'mm-components ') => 'default',
+                    __( 'Ghost (transparent background, white border)', 'mm-components ') => 'ghost',
+                ),
+            ),
+            array(
+                'type' => 'dropdown',
+                'heading' => __( 'Button Color', 'mm-components' ),
+                'param_name' => 'button_color',
+                'value' => array(
+                    __( 'Default', 'mm-components ') => 'default',
+                    __( 'Pink', 'mm-components ') => 'pink',
+                    __( 'White', 'mm-components ') => 'white',
+                    __( 'Gray', 'mm-components ') => 'gray',
+                ),
+                'dependency' => array(
+                    'element' => 'button_style',
+                    'value' => array(
+                        'ghost',
+                    ),
+                ),
+            ),
+            array(
+                'type' => 'textarea_raw_html',
+                'heading' => __( 'Secondary Call to Action', 'mm-components' ),
+                'param_name' => 'secondary_cta',
+                'description' => __( 'Outputs below the main button, can include HTML markup.', 'mm-components' ),
+            ),
+        ),
+    ) );
+}
+
+add_action( 'widgets_init', 'mm_components_register_hero_banner_widget' );
+/**
+ * Register the widget.
+ *
+ * @since  1.0.0
+ */
+function mm_components_register_hero_banner_widget() {
+    register_widget( 'mm_hero_banner_widget' );
+}
+/**
+ * Hero Banner Widget.
+ *
+ * @since  1.0.0
+ */
+class Mm_hero_banner_Widget extends Mm_Components_Widget {
```

Comment:
This should be Mm_Hero_Banner_Widget (all caps for first letters in class names)